app/subscriber/src/features/utils/createSubMediaGroups.ts

```diff
@@ -0,0 +1,60 @@
+import {
+  ISubMediaGroupExpanded,
+  ISubMediaGroupItem,
+} from 'features/search-page/components/advanced-search/interfaces';
+import { IMediaTypeModel, ISourceModel } from 'tno-core';
+
+/**
+ * Create Sub Media Groups
+ * @param sources
+ * @param mediaTypes
+ * @param setSubMediaGroups
+ * @param setMediaGroupExpanded - optional
+ */
+export const createSubMediaGroups = (
```

Comment:
This is a perfect example of something that should become a hook.  If you're passing state stuff from other components, just move the state stuff to the hook so that the component can pull it out instead like all other hooks.

Reply:
Just debugging some dependency bug with the new hook, will ping once I fix it

Reply:
New commit [here!](https://github.com/bcgov/tno/pull/1533/commits/0503937675c1a01028e9f0795215a27dae802995) @Fosol